File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        unordered_map<char, int> map;
        int ans = INT_MIN;
        
        int start = 0;
        for(int end=0; end<s.size(); end++) {
            map[s[end]]++;
            
            while(start <= end && map[s[end]] > 2) {
                map[s[start]]--;
                start++;
            }
            
            ans = max(ans, end - start + 1);
        }
        
        return ans;
    }
};

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int[] map = new int[26];
        int maxLen = 0;
        int i = 0;
        for(int j=0; j<s.length(); j++) {
            char ch = s.charAt(j);
            map[ch - 'a'] += 1;
            while(map[ch - 'a'] > 2) {
                map[s.charAt(i) - 'a'] -= 1;
                i++;
            }
            maxLen = Math.max(maxLen, j - i + 1);
        }
        return maxLen;
    }
}